crates/astria-sequencer-relayer/src/main.rs

```diff
@@ -44,10 +51,30 @@ async fn main() -> ExitCode {
         "initializing sequencer relayer"
     );
 
-    SequencerRelayer::new(cfg)
-        .expect("could not initialize sequencer relayer")
-        .run()
-        .await;
+    let mut sigterm = signal(SignalKind::terminate())
+        .expect("setting a SIGTERM listener should always work on Unix");
+    let (sequencer_relayer, shutdown_handle) =
+        SequencerRelayer::new(cfg).expect("could not initialize sequencer relayer");
+    let sigterm_handle = tokio::spawn(async move {
```

Comment:
Does it make sense to invert this, i.e. `spawn(sequencer_relayer.run()` instead of the signal handlers?

Reply:
Yup - sure.